File: Data Structures & Algorithms/merge-intervals/submission-10.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        res = []
        intervals.sort()
        new = intervals[0] #initiate the new interval

        for i in intervals[1:]:
            if i[0] <= new[1]:
                new[1] = max(i[1], new[1])
                new[0] = min(i[0], new[0])
            else:
                res.append(new)
                new = i
            
        res.append(new)

        return res
      

File: Data Structures & Algorithms/maximum-product-subarray/submission-10.java
class Solution {
    public int maxProduct(int[] nums) {
        int curMax = nums[0], curMin = nums[0], res = nums[0];
        for(int i = 1; i < nums.length; i++){
            int tempMax = curMax;
            curMax = Math.max(nums[i], Math.max(tempMax*nums[i], curMin*nums[i]));
            curMin = Math.min(nums[i], Math.min(tempMax*nums[i], curMin*nums[i]));
            res = Math.max(curMax, res);
        }
        return res;
    }
}
